storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestConstraintSolverStrategy.java

```diff
@@ -210,7 +210,7 @@ protected SolverResult backtrackSearch(SearcherState state) {
      * Cluster has sufficient resources for scheduling to succeed but can fail due to StackOverflowError.
      */
     @ParameterizedTest
-    @ValueSource(ints = {5, 20})
+    @ValueSource(ints = {1, 20})
```

Comment:
I would also update L241
 https://github.com/apache/storm/blob/master/storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestConstraintSolverStrategy.java#L241-L248 to reflect the change. 

Maybe change to 

```
if (parallelismMultiplier <= 1) {
   xxxx;
} else {
   yyyy;
}
```